lightningd/bolt11.c

```diff
@@ -248,6 +248,33 @@ static char *decode_x(struct bolt11 *b11,
         return NULL;
 }
 
+/* BOLT #11:
+ *
+ * `c` (24): `data_length` variable.  `min_final_cltv_expiry` to use for the
+ * last HTLC in the route. Default is 9 if not specified.
+ */
+#define DEFAULT_C 9
+static char *decode_c(struct bolt11 *b11,
+                      struct hash_u5 *hu5,
+                      u5 **data, size_t *data_len,
+                      size_t data_length, bool *have_c)
+{
+        u64 c;
+        if (*have_c)
+                return unknown_field(b11, hu5, data, data_len, 'c',
+                                     data_length);
+
+        /* FIXME: Put upper limit in bolt 11 */
+        if (!pull_uint(hu5, data, data_len, &c, data_length * 5))
+                return tal_fmt(b11, "c: length %zu chars is excessive",
+                               *data_len);
+        b11->min_final_cltv_expiry = c;
+        if (b11->min_final_cltv_expiry != c)
```

Comment:
How could this ever happen? We set it to be equal above

Reply:
Different types, could overflow. I'll add a comment.